onnxruntime/core/framework/kernel_registry.cc

```diff
@@ -309,6 +309,7 @@ Status KernelRegistry::TryFindKernel(const Node& node,
         << " kernel is not supported in " << expected_provider << "."
         << " Encountered following errors: (" << ToString(verify_kernel_def_error_strs) << ")";
 
+    VLOGS_DEFAULT(2) << "TryFindKernel failed, Reason: " << oss.str();
```

Comment:
It is misleading. This function needs be call again and again. If the first call failed, then a second one passed. Here you will print out an error but normally end users shouldn't see such internal details and they don't care how many searches have been made to find a suitable one.

Reply:
The end user will not see it. `VLOGS_` is only enabled in Debug build. This is kernel developer oriented.